core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -1338,16 +1338,12 @@ class AdminClientIntegrationTest extends IntegrationTestHarness with Logging {
     changePreferredLeader(prefer1)
     // but shut it down...
     servers(1).shutdown()
-    waitUntilTrue (
-      () => {
-        val description = client.describeTopics(Set (partition1.topic(), partition2.topic()).asJava).all().get()
-        return !description.asScala.flatMap{
-          case (topic, description) => description.partitions().asScala.map(
-            partition => partition.isr().asScala).flatten
-        }.exists(node => node.id == 1)
-      },
-      "Expect broker 1 to no longer be in any ISR"
-    )
+    waitUntilTrue (() => {
+      val description = client.describeTopics(Set(partition1.topic, partition2.topic).asJava).all().get()
+      !description.asScala.flatMap { case (topic, description) =>
+        description.partitions.asScala.flatMap(_.isr.asScala)
+      }.exists(_.id == 1)
+    }, "Expect broker 1 to no longer be in any ISR")
```

Comment:
@tombentley after I fixed this, the test started failing. `return` was causing everything after this not to run.

Reply:
@ijuma : Good find. These are indeed existing bugs. There are 4 references of LeaderNotAvailableException in line 1358, 1372, 1380 and 1390. They all need to be changed to PreferredLeaderNotAvailableException. Once I fixed those locally, the test passes.

Reply:
Thanks Jun! Will fix.

Reply:
Thanks Jun! Will fix.